-1

Tested at: 7622e302d0452e1dcfc67170d3b2605d9f008c85

CMSSW: CMSSW_11_2_X_2020-07-22-1200
SCRAM_ARCH: slc7_amd64_gcc820

The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
 - #30478 @christopheralanwest: Update global tags to use V5 JEC for 2017 data/MC and Run 3 MC and add V7b JR for 2018 data/MC
 - #30852 @andrius-k: Added links to inspect bin by bin comparison with a raw rootjs viewer
 - #30865 @vargasa: DD4hep: SimTracker/TrackerMaterialAnalysis XML Location
 - #30812 @JamminJones: added esConsumes to  modules in RecoTracker/FinalTrackSelectors

You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-c39730/8229/git-log-recent-commits
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-c39730/8229/git-merge-result

You can see the results of the tests here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-c39730/8229/summary.html

I found follow errors while testing this PR

**Failed tests:  Build  HeaderConsistency**

* **Build**:

I found compilation error when building: 

<pre>>> Compiling  /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_2_X_2020-07-22-1200/src/FastSimulation/Event/src/FSimVertex.cc
>> Compiling  /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_2_X_2020-07-22-1200/src/FastSimulation/Event/src/KineParticleFilter.cc
In file included from /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_2_X_2020-07-22-1200/src/DataFormats/HepMCCandidate/interface/GenParticleFwd.h:6,
                 from /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_2_X_2020-07-22-1200/src/FastSimulation/Event/interface/FSimEvent.h:6,
                 from /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_2_X_2020-07-22-1200/src/FastSimulation/Event/src/FSimEvent.cc:2:
/data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_2_X_2020-07-22-1200/src/DataFormats/Common/interface/FwdRef.h:129:29: error: 'remove_cv' in namespace 'boost' does not name a template type
     typedef typename boost::remove_cv<typename boost::remove_reference<argument_type>::type>::type key_type;
                             ^~~~~~~~~
/data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_2_X_2020-07-22-1200/src/DataFormats/Common/interface/FwdRef.h:129:38: error: expected unqualified-id before '<' token
     typedef typename boost::remove_cv<typename boost::remove_reference<argument_type>::type>::type key_type;
                                      ^
</pre>


The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
 - #30478 @christopheralanwest: Update global tags to use V5 JEC for 2017 data/MC and Run 3 MC and add V7b JR for 2018 data/MC
 - #30852 @andrius-k: Added links to inspect bin by bin comparison with a raw rootjs viewer
 - #30865 @vargasa: DD4hep: SimTracker/TrackerMaterialAnalysis XML Location
 - #30812 @JamminJones: added esConsumes to  modules in RecoTracker/FinalTrackSelectors

You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-c39730/8229/git-log-recent-commits
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-c39730/8229/git-merge-result

--- a/DataFormats/Common/interface/FwdRef.h
+++ b/DataFormats/Common/interface/FwdRef.h
@@ -113,6 +113,8 @@ FwdRef: A template for a interproduct reference to a member of a product.
 #include "DataFormats/Common/interface/CMS_CLASS_VERSION.h"
 #include "DataFormats/Common/interface/Ref.h"
 
+#include <boost/functional.hpp>
+
 namespace edm {
 
   template <typename C,
@@ -126,7 +128,7 @@ namespace edm {
     typedef T const element_type;  //used for generic programming
     typedef F finder_type;
     typedef typename boost::binary_traits<F>::second_argument_type argument_type;
-    typedef typename boost::remove_cv<typename boost::remove_reference<argument_type>::type>::type key_type;
+    typedef typename std::remove_cv<typename std::remove_reference<argument_type>::type>::type key_type;
     /// C is the type of the collection
     /// T is the type of a member the collection
 

--- a/L1Trigger/TrackFindingTMTT/interface/TrackerModule.h
+++ b/L1Trigger/TrackFindingTMTT/interface/TrackerModule.h
@@ -10,6 +10,7 @@
 #include <set>
 #include <array>
 #include <map>
+#include <cmath>
 
 class TrackerGeometry;
 class TrackerTopology;

--- a/RecoLocalCalo/HcalRecAlgos/interface/HBHEPulseShapeFlag.h
+++ b/RecoLocalCalo/HcalRecAlgos/interface/HBHEPulseShapeFlag.h
@@ -14,6 +14,7 @@
 #include <string>
 #include <vector>
 #include <map>
+#include <cmath>
 //---------------------------------------------------------------------------
 #include "DataFormats/HcalDigi/interface/HBHEDataFrame.h"
 #include "DataFormats/HcalRecHit/interface/HcalRecHitCollections.h"

